docs/Using Randomness/chainlink-vrf.md

```diff
@@ -35,110 +35,47 @@ Use Chainlink VRF to build reliable smart contracts for any applications that re
+Chainlink VRF v2 offers two methods for requesting randomness:
 
-These variables depend on current network conditions, your specified limit on callback gas, and the number of random values in your request. The cost of each request is final only after the transaction is complete, but you define the limits you are willing to spend for the request with the following variables:
+- [Subscription](/docs/vrf/v2/subscription/): Users create a subscription account and fund its balance with LINK tokens. Users can then link one or several consuming contracts to the subscription account. When the consuming contracts request randomness, the transaction costs are calculated once the randomness requests are fulfilled and the subscription balance is deducted accordingly. This mode allows the users to fund their use of Chainlink VRF from a single location.
+- [Ad-hoc](/docs/vrf/v2/ad-hoc/): Consuming contracts directly pay with LINK when they request for randomness. Hence users have to directly fund their consuming contracts and ensure there are enough LINK tokens to pay for randomness requests.
 
-- **Gas lane:** The maximum gas price you are willing to pay for a request in wei. Define this limit by specifying the appropriate `keyHash` in your request. The limits of each gas lane are important for handling gas price spikes when Chainlink VRF bumps the gas price to fulfill your request quickly.
+## Choosing the right method
 
-- **Callback gas limit:** Specifies the maximum amount of gas you are willing to spend on the callback request. Define this limit by specifying the `callbackGasLimit` value in your request.
+Depending on your use case, one method might be more suitable than another. The recommendations are the following:
 
-Requests to Chainlink VRF v2 follow the [Request & Receive Data](/docs/request-and-receive-data/) cycle. The VRF coordinator processes the request and determines the final charge to your subscription using the following steps:
+- If your use case requires to regularly request for randomness then choose the subscription method. Otherwise , choose the ad-hoc method. The ad-hoc method is more suitable to one-off/seldom requests.
+- If you have several VRF consuming contracts then choose the subscription method.
+- If you want more control on the maximum gas price to be paid and less gas overhead then choose the subscription method. You can read the [Subscription](/docs/vrf/v2/subscription/) and [Ad-hoc](/docs/vrf/v2/ad-hoc/) explanation pages to understand how the transaction costs are calculated.
+- Due to an overhead with the ad-hoc method, you can request much more random words in a single request with the Subscription method than the Ad-hoc method. You can compare the maximum number of words per request and per method: [Subscription configuration](/docs/vrf/v2/subscription/configuration/#configurations) and [Ad-hoc configuration](/docs/vrf/v2/ad-hoc/configuration/#configurations).
```

Comment:
```suggestion
- If your use case requires regular requests for randomness, choose the subscription method to simplify funding and reduce the overall cost. Otherwise , choose the ad-hoc method. The ad-hoc method is more suitable for infrequent one-off requests.
- If you have several VRF consuming contracts, choose the subscription method.
- To reduce gas overhead and have more control over the maximum gas price for requests, choose the subscription method. You can read the [Subscription](/docs/vrf/v2/subscription/) and [Ad-hoc](/docs/vrf/v2/ad-hoc/) explanation pages to understand how the transaction costs are calculated.
- Because the ad-hoc method has higher overhead, it cannot return as many random words in a single request as the subscription method. You can compare the maximum number of words per request and per method on the [Subscription configuration](/docs/vrf/v2/subscription/configuration/#configurations) and [Ad-hoc configuration](/docs/vrf/v2/ad-hoc/configuration/#configurations) pages.
```